docs/development/javascript/how-to-build-and-use-vuejs-components/index.md

```diff
@@ -2,388 +2,561 @@
+                </div>`
+})
+
+Vue.component('rating-title', {
+    template:   `<div>
+                    <h1>
+                        <slot>Default Rating Title</slot>
+                    </h1>
+                </div>`
+})
+
+new Vue({
+    el: '#app',
+    data: {
+        parentHeader: {
+            label: "Counter is at: ",
+            globalCount: 0
+        }
+    }
+})
 </script>
-```
-You can see that within our rating counter component there are now two slots, but this time they have name attributes as well ```<slot name="title"></slot>``` and respectively ```<slot name="subtitle"></slot>```. This takes care of the creation of the named slots, simple right? But how do we use it? You can reference a named slot by using the ```<template v-slot:slotname>``` within your parent component:
-```html
-<rating-counter v-bind:parent="parentTitle">
-       <template v-slot:title>
-           <rating-title></rating-title>
-       </template>
-       <template v-slot:subtitle>
-           <h2>Subtitle</h2>
-       </template>
-   </rating-counter>
-```
+{{< /file >}}
 
-## What are component events
-Events in components are essential to component -> parent communication in VueJS. Whenever a component needs to communicate back to its parent this is the safe way of doing that. I've built a simplified example:
-```html
+When loaded in a browser, the file will look like:
+
+![Components with named slots](component-named-slots.png "Components with named slots")
+
+-   You can see that within our `rating-counter` component (on lines 26-29) there are now two slots. This time they have name attributes as well: `title` and `subtitle`.
+
+-   This takes care of the creation of the named slots, but how do we use them? You can reference a named slot by using the `<template v-slot:slotname></template` syntax within your parent component. The content inside the `<template>` tags will then be inserted where the `<slot>` with the same name appears in your component's template.
+
+## Using Component Events
+
+[Events](https://vuejs.org/v2/guide/components.html#Listening-to-Child-Components-Events) in components are essential for component-to-parent communication in VueJS. Whenever a component needs to communicate back to its parent, this is the safe way of doing that.
+
+For an example of how this works, replace the contents of your `ratingcounter.html` with the following snippet:
+
+{{< file "ratingcounter.html" html >}}
 <script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>
 
 <div id="app">
-    {{ parentTitle.title+" "+parentTitle.counter }}
-    <rating-counter v-bind:parent="parentTitle" v-on:increment="parentTitle.counter++"></rating-counter>
+    {{ parentHeader.label+" "+parentHeader.globalCount }}
+    <hr />
+
+    <rating-counter v-bind:parent="parentHeader" v-on:increment="parentHeader.globalCount++"></rating-counter>
 </div>
+
 <script>
-    Vue.component('rating-counter', {
-        props: ['parent'],
```

Comment:
Removed the parent prop and the component data, as they were no longer being used.